src/integration/java/io/pinecone/integration/dataPlane/UpsertDescribeIndexStatsAndDeletePodTest.java

```diff
@@ -23,56 +23,59 @@
 
 public class UpsertDescribeIndexStatsAndDeletePodTest {
-    private static Pinecone pineconeClient;
-    private static String indexName;
-    private static Index indexClient;
-    private static AsyncIndex asyncIndexClient;
+    private static Index index;
+    private static AsyncIndex asyncIndex;
     private static final int dimension = 3;
     private static final Struct nullFilterStruct = null;
 
     @BeforeAll
     public static void setUp() throws IOException, InterruptedException {
         AbstractMap.SimpleEntry<String, Pinecone> indexAndClient = createIndexIfNotExistsDataPlane(dimension, IndexModelSpec.SERIALIZED_NAME_POD);
-        indexName = indexAndClient.getKey();
-        pineconeClient = indexAndClient.getValue();
-        indexClient = pineconeClient.createIndexConnection(indexName);
-        asyncIndexClient = pineconeClient.createAsyncIndexConnection(indexName);
+        String indexName = indexAndClient.getKey();
+        Pinecone pineconeClient = indexAndClient.getValue();
+        index = pineconeClient.createIndexConnection(indexName);
+        asyncIndex = pineconeClient.createAsyncIndexConnection(indexName);
     }
 
     @Test
     public void upsertVectorsAndDeleteByIdSyncTest() throws InterruptedException {
         // Upsert vectors with required parameters
+        int dimension = 3;
+        Struct emptyFilterStruct = null;
         int numOfVectors = 3;
+
         String namespace = RandomStringBuilder.build("ns", 8);
         List<String> upsertIds = getIdsList(numOfVectors);
-        int vectorCount = 0;
         for (String id : upsertIds) {
-            UpsertResponse upsertResponse = indexClient.upsert(id,
+            index.upsert(id,
                     generateVectorValuesByDimension(dimension),
                     namespace);
-            vectorCount += upsertResponse.getUpsertedCount();
         }
 
-        int actualVectorCount = vectorCount;
+        int actualVectorCount = numOfVectors;
+
+        // wait sometime for the vectors to be upserted
+        Thread.sleep(10000);
 
         assertWithRetry(() -> {
             // call describeIndexStats to get updated vector count
-            DescribeIndexStatsResponse describeIndexStatsResponse = indexClient.describeIndexStats(null);
+            DescribeIndexStatsResponse describeIndexStatsResponse = index.describeIndexStats();
 
             // verify the updated vector count
             assertEquals(describeIndexStatsResponse.getNamespacesMap().get(namespace).getVectorCount(), actualVectorCount);
         });
 
         // Delete 1 vector
         List<String> idsToDelete = new ArrayList<>(3);
-        idsToDelete.add(upsertIds.get(0));
-        indexClient.delete(idsToDelete, false, namespace, nullFilterStruct);
-        vectorCount -= idsToDelete.size();
-        int testSingleDeletedVectorCount = vectorCount;
+        String vectorIdToDelete = upsertIds.get(0);
+        idsToDelete.add(vectorIdToDelete);
```

Comment:
Why the change pulling out `vectorIdToDelete`?

Reply:
We are just assigning upsertIds.get(0) to vectorIdToDelete which is still added to idsToDelete.